spec/Order.md

```diff
@@ -0,0 +1,58 @@
+CodeChain supports an on-chain order to support DEX(Decentralized Exchange).
+
+## Order
+
+An order can be either a _point-to-point order_ between a maker and a taker, or a _broadcast order_ between a maker, a relayer and takers.
+Point-to-point orders allow two parties(makers and takers) to directly exchange assets between each other.
+Broadcast orders are similar to point-to-point orders, but also allows relayers to take fee assets from makers and takers. Broadcast orders are usually used in decentralized exchange platforms.
+
+Orders on CodeChain is implemented on UTXO forms.
+Basically, an order is like a tag on products in grocery stores. Orders can be put on inputs and outputs of transfer transactions. If an input/output has an order, it means that specific input/output should be exchanged as its order says. Think about this situation: Let’s say we have an order to exchange 10 gold to 100 silver, and we've put that order on a 10-gold input. Then there should be a 100-silver output with the same order of 10 gold to 100 silver. If there isn’t enough gold, there would perhaps be a 5-gold output and a 50-silver output, both with the same order, or, whatever outputs with the same order while maintaining the order equilibrium.
+
+Assets with orders must be able to be spent by takers or relayers without any permission. But in the CodeChain UTXO form, those parties should provide a lock script and an unlock script to prove the ownership of the assets. To solve the problem, CodeChain runs VM on **orders**, not *transactions*, if there are orders on inputs. If there are no orders on inputs, CodeChain runs VM on transactions as usual. Partial signing is not allowed on transactions with orders. With this convention, takers and relayers can take the ownership of the assets with orders with unlock scripts which are provided by makers.
```

Comment:
To solve the problem, CodeChain runs VM on **orders**, not *transactions*, if there are orders on inputs. -> In order to solve the problem, if there are orders on inputs, CodeChain runs VM on **orders**, not *transactions*.